awsshell/config.py

```diff
@@ -75,9 +75,17 @@ def _copy_template_to_config(self, template_path,
         :type overwrite: bool
         :param overwrite: (Optional) Determines whether to overwrite the
             existing config file, if it exists.
+
+        :raises: :class:`OSError <exceptions.OSError>`
         """
         expanded_config_path = os.path.expanduser(config_path)
         if not overwrite and os.path.isfile(config_path):
             return
         else:
+            try:
+                config_path_dir_name = os.path.dirname(config_path)
+                os.makedirs(config_path_dir_name)
```

Comment:
We might want to consolidate the places where we create the ~/.aws/shell/ directory (and sub directories).  I count four separate locations now in the codebase (including this new call) where we are calling `os.makedirs`.